src/Castle.Core/DynamicProxy/Contributors/ClassProxySerializableContributor.cs

```diff
@@ -187,7 +191,11 @@ private bool VerifyIfBaseImplementsGetObjectData(Type baseType, IList<MethodInfo
 				throw new ArgumentException(message);
 			}
 
-			methodsToSkip.Add(getObjectDataMethod);
+			if (model.Methods.FirstOrDefault(m => m.Method == getObjectDataMethod) is { } method &&
```

Comment:
First, note that this line of code goes away in a later commit so any changes will only remain in a historical commit.

That said, I'm happy to change to a different syntax. I don't yet know your preference regarding pattern matching syntax, so this is probably a good opportunity to briefly discuss it.

I chose the syntax for brevity. I needed a way to name the queried `MetaMethod` because it's subsequently needed (`method.Ignore = true`), thus the `is … method` syntax. I chose `is { }` over `is MetaMethod` for brevity, and I chose it over `is var` for correctness as the latter also matches `null`.

* The most terse syntax would be (I think) `… is { Proxyable: true } method`. I didn't think of that at the time.
* The classic way would be: `var method = model.Methods.FirstOrDefault(…); if (method != null && method.Proxyable) …` or `… if (method?.Proxyable == true)`.

Let me know which you prefer.

Reply:
Oops, yer I meant to remove that comment, I had added it on a commit but it didn't show up when I got to the full diff to remove it.

I haven't used pattern matching much, too many clients still with either older toolsets or teams not using it, so I've avoided being fancy. I see exactly why you used it now, thanks for the explanation, I usually put the Linq extension methods query into a local on the line before an if-statement (the classic way) because the intent is clear especially because the lambda makes the if-statement already long, but I'm sure I could get used to this.